services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusUpdateRequest.kt

```diff
@@ -12,12 +12,12 @@ import java.time.ZonedDateTime
 import java.util.UUID
 
 sealed interface HakemusUpdateRequest {
-    val name: String
-    val postalAddress: PostalAdressRequest
-    val startTime: ZonedDateTime?
-    val endTime: ZonedDateTime?
-    val areas: List<ApplicationArea>?
-    val customerWithContacts: CustomerWithContactsRequest?
+    var name: String
```

Comment:
Why were all fields changed to vars? There's no reason for the request to change after creation.

Reply:
This is also something I did for testing purposes where I create the request from Hakemus data object and then change something in it. I could use vals and copy instead but I had trouble figuring out how to do it with nested fields, e.g. to change HakemusUpdateRequest.customerWithContacts.contact.id. using copy. Maybe it is easy and I have to just google a bit moore for the solution.

Reply:
Using copy on nested objects, especially deeply nested objects, can be annoying. But it can be made less annoying by using extension methods. It's also often easier to create the right kind of nested object than it is to modify an existing one.

Reply:
Still trying to figure out a proper extension function mechanism for copy. But in the mean time I reverted the fields as vals.